src/commandlineflags.cc

```diff
@@ -216,4 +209,34 @@ bool ParseStringFlag(const char* str, const char* flag, std::string* value) {
 bool IsFlag(const char* str, const char* flag) {
   return (ParseFlagValue(str, flag, true) != nullptr);
 }
+
+// If |*pstr| begins with |prefix|, advance *pstr so that it points
+// just past the prefix, and return true. Otherwise, return false.
+static bool SkipPrefix(const char* prefix, const char** pstr) {
+  assert(prefix);
+  assert(pstr);
+  size_t prefix_len = strlen(prefix);
+  if (0 == strncmp(*pstr, prefix, prefix_len)) {
+    *pstr += prefix_len;
+    return true;
+  }
+  return false;
+}
+
+bool HasBenchmarkFlagPrefix(const char* str) {
+  assert(str);
+  return (SkipPrefix("--", &str) ||
+          SkipPrefix("-", &str) ||
+          SkipPrefix("/", &str)) &&
+         (SkipPrefix("benchmark_", &str) ||
+          SkipPrefix("benchmark-", &str));
+}
+
+bool IsTruthyFlagValue(const char* value)
```

Comment:
Is there a reason this one should take a std::string when the others in the header take a c string?